### ISSUE TITLE
-Set the serverless env var for staging

### ISSUE THREAD

**Issue by ShanaLMoore** [scientist-softserv/britishlibrary#336]
# Summary

dependent on #335 

- [x] Update BL's iiif_print to pull in changes and test. Done by #381 
- [x] Set the serverless env var for staging (EXTERNAL_IIIF_URL)

## Acceptance Criteria 

- [ ] Images should get served from cloudfont. Inspect the iframe and confirm the url no longer points to the rails app.

# Screenshots or Video
https://d2ltm8fy0v0agc.cloudfront.net/iiif/2/b4543de001a313a65b1a01d72356c4b815d2e47b/full/600,/0/default.jpg

# Testing Instructions
- [ ] QA will probably need to pair with a Dev. 
- [ ] We will need to inspect element, or visit /manifest.json of a work, to review the image's url paths. The domain should be cloudfront...
- [ ] client QA is not required

# Notes
- [ ] Confirm if we'll need to reindex staging after these changes.

**Comment by jillpe** [comment]
Blocked: waiting for Cloud URL

**Comment by ShanaLMoore** [comment]
unblocked. Rory provided the following cloudfront url

**Comment by kirkkwang** [comment]
Sounds good, is it safe to push to staging yet?

**Comment by ShanaLMoore** [comment]
> Sounds good, is it safe to push to staging yet?

I think so since Jeremy deployed it to prod, but please confirm with Jeremy first anyways. ref: https://assaydepot.slack.com/archives/C0313NK2LJ0/p1680099034697759

**Comment by ShanaLMoore** [comment]
marking as blocked again. We can complete this once the cloudfront url is in place and working. ref: #334

**Comment by ShanaLMoore** [comment]
ref slack convo: https://assaydepot.slack.com/archives/G0313NK5NMA/p1680267231307789

**Comment by ShanaLMoore** [comment]
Rob was able to get this working with the following. 

https://d2ltm8fy0v0agc.cloudfront.net/iiif/2/b4543de001a313a65b1a01d72356c4b815d2e47b/full/600,/0/default.jpg 

I asked if there's a diff uri to set prod: https://assaydepot.slack.com/archives/G030UPFBT2S/p1681483795858109?thread_ts=1681418361.087269&cid=G030UPFBT2S

**Comment by ShanaLMoore** [comment]
**INTERNAL QA: **

Confirmed that EXTERNAL_URL was set in BL staging: 

<img width="436" alt="image" src="https://user-images.githubusercontent.com/10081604/232606675-9ce81a3b-59c9-48fa-94c7-7a6fe521e2a2.png">

Confirmed manifest references cloudfront url:

**Comment by ShanaLMoore** [comment]
While QA-ing another ticket, we noticed that only the BL tenant is misbehaving. The BL tenant will not render any images in the UV. 



![Image](https://user-images.githubusercontent.com/10081604/232629972-26406292-668f-4798-9188-3f45bee7022b.png)

Visiting the [cloudfront urls](https://d2ltm8fy0v0agc.cloudfront.net/iiif/2/cf7a4206c7caec31f124aef0bb65b94dbab6ca5c/full/600,/0/default.jpg) directly produce 404 errors: 



![Image](https://user-images.githubusercontent.com/10081604/232630105-44aefff4-2195-4b89-b606-56866d86725c.png)

**Comment by cziaarm** [comment]
Odd one, I tried adding that cat again and it doesn't (or at least hasn't yet) show(n) up in the bucket: 

Here's what I found when comparing records for viral_snoopy.jpg and my latest attempt to upload a stary cat:

The file metadata for Viral snoopy record in fedora has the digest that matches an object in the  bl-staging-fcrepo bucket

**http://localhost:8889**/rest/af879637-eeeb-4fbe-802b-3999e8e1eb26/17/92/69/e0/179269e0-5017-4b7d-91bc-cc302979e03d/files/b238c03a-172f-4cb3-8046-3448ac57d3b6/fcr:metadata

premis: hasMessageDigest
urn:sha1:3a8e1e7931af6ec168a9d993f987138f3b86c759

This has a corresponding object in the bucket:

https://s3.console.aws.amazon.com/s3/object/samvera-fcrepo-staging-bl?region=eu-west-1&prefix=3a8e1e7931af6ec168a9d993f987138f3b86c759

The stary cat  (360_F_97589769_t45CqXyzjz0KXwoBZT9PRaWGHRk5hQqQ.jpg)

fcrepo metadata: http://localhost:8889/rest/af879637-eeeb-4fbe-802b-3999e8e1eb26/89/99/8d/2b/89998d2b-b7d7-4f46-a1dd-71884097ce8c/files/6557b9aa-4b79-40b6-a520-8d6a21dbedba/fcr:metadata

premis: hasMessageDigest
urn:sha1:cf7a4206c7caec31f124aef0bb65b94dbab6ca5c

but no object in bucket! https://s3.console.aws.amazon.com/s3/object/samvera-fcrepo-staging-bl?region=eu-west-1&prefix=cf7a4206c7caec31f124aef0bb65b94dbab6ca5c

I realise that this is just a fancy way of confirming the 404 message we are seeing and doesn't shed any light on why...

My understanding is that fcrepo is configured to store stuff in the bucket. It is storing the cat pic (I can download it from hyku and direct from fedora), but why not appearing in the bucket? 

I checked in case a miss-config meant that staging content was going into the prod bucket, but that doesn't appear to be the case.

**Comment by cziaarm** [comment]
@jillpe @ShanaLMoore I was going to draw attention to this ^^ on slack, but slack won't start this afternoon :/ 

I'd love to find out more about how fcrepo is configured to use the s3 bucket. It would seem that somehow it is not doing that at the moment, or at least not in a way that is consistent with the bitstream IDs that are in fedora.

**Comment by cziaarm** [comment]
Thanks @ShanaLMoore I did find that PR and double checked to see if those changes hadn't gotten reverted in main, which it doesn't look like they have.

It is odd as some of those SHAs evidently do match. 

If you upload a new image to bl staging does your image get a 404 too? I may be being paranoid, but it looks like all the ones I've done don't work and the ones that you had done do... maybe fcrepo has the hump with me 😂

**Comment by cziaarm** [comment]
Here's a thing though ops/provision/k8s/fcrepos3-staging-values.yaml which I think is referenced here https://github.com/scientist-softserv/britishlibrary/pull/337/files#diff-82b3e9d6fc63de045ee16f94d51b1dcd76c4c782d8ffeb04a28a10f8da2c0b92R172 does not exist on main. perhaps significant!

**Comment by cziaarm** [comment]
Mola cat: https://mola.bl-staging.notch8.cloud/uv/uv.html#?manifest=https://mola.bl-staging.notch8.cloud/concern/images/d1cefbf1-ee22-4af6-956f-845fe24bd669/manifest&config=https://mola.bl-staging.notch8.cloud/uv/uv-config.json

**Comment by cziaarm** [comment]
@ShanaLMoore The cat came back!

https://d2ltm8fy0v0agc.cloudfront.net/iiif/2/cf7a4206c7caec31f124aef0bb65b94dbab6ca5c/full/600,/0/default.jpg

https://www.youtube.com/watch?v=Ck0jwS0CvKk

**Comment by ShanaLMoore** [comment]
Ya i see now! Sorry I must've missed this before. It looks like it's working now. I see all my snoopys again
